### PR TITLE
feat: branding and marketing related updates

### PR DESCRIPTION
**Pull request by SiTaggart**
This pull request includes changes mainly related to design tokens and UI components in the Twilio Paste library. The most significant changes are the adjustments to the bottom margin of the heading variant, the addition of new branding accent background colors and a new text color token, and the addition of new icons.

Design token updates:

* [`.changeset/four-walls-lick.md`](diffhunk://#diff-9f2a9955bb8538af52e124ad85f62e60ceca07f4b4d85cfa95ae469dc5eaa79dR1-R6): Patched the `@twilio-paste/heading` and `@twilio-paste/core` packages with a minor adjustment to the bottom margin of the heading variant 20.
* [`.changeset/swift-lobsters-prove.md`](diffhunk://#diff-cfed4caf83c716d7267aaa28ed26d0fd7225f0620c13efc86f42ae4c7e42b0c6R1-R6): Updated the `@twilio-paste/design-tokens` and `@twilio-paste/core` packages with three new branding accent background colors and a new text color token.
* [`packages/paste-design-tokens/tokens/global/background-color.yml`](diffhunk://#diff-1d81e5cf80ef27ace941e8aeaaa787a2774303a1bc802c1da02d43201fe2bd0cR53-R61): Added new branding accent background colors representing Twilio, Sendgrid, and Segment.
* [`packages/paste-design-tokens/tokens/global/text-color.yml`](diffhunk://#diff-ce72df3e84cab8ae3ef60acc32ba9e59d498299c05874b0f60868bcc57d965baR84-R90): Added a new text color token to be used on top of the new branding accent background colors.
* `packages/paste-design-tokens/tokens/themes/twilio-dark/global/background-color.yml` and `packages/paste-design-tokens/tokens/themes/twilio/global/background-color.yml`: Updated 3 branding background colors in the Twilio and twilio-dark themes. [[1]](diffhunk://#diff-ba10c929247437f671c794a0267712565e5b1d8a5357ff770f1c21c971343e23R15-R25) [[2]](diffhunk://#diff-706f7b1540e23c479e1624ba5f97901a7aecb346c98c05b97f688620092a2358R16-R26)

UI component updates:

* `packages/paste-core/components/heading/__tests__/heading.test.tsx` and `packages/paste-core/components/heading/src/Heading.tsx`: Adjusted the bottom margin of the heading variant 20 from `1.5rem` to `1.25rem`. [[1]](diffhunk://#diff-606f768b3568aa357220f64db3d3a84e92d74487f9acff323715bc0f1b7e36cdL36-R36) [[2]](diffhunk://#diff-606f768b3568aa357220f64db3d3a84e92d74487f9acff323715bc0f1b7e36cdL118-R118) [[3]](diffhunk://#diff-935899a6ad77bf56a0d1cc270ed7261d7db4ee284702a21c1faf2ab704634e55L56-R56)
* [`packages/paste-icons/build.icon-list.js`](diffhunk://#diff-738590665dcbaae46090cf672f8914889bb5603a84a25dc1a1dbf0f98a3243c9L1-R1): Added `ArtificialIntelligenceIcon`, `AvailableIcon`, and `ProductSendGridIcon` to the list of icons.

### CODE REVIEW

File: .changeset/thirty-poets-notice.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@twilio-paste/icons": minor
```
Comment (nkrantz) [inline .changeset/thirty-poets-notice.md:2]: include core?

**Review by nkrantz — DISMISSED** [review]

File: .changeset/thirty-poets-notice.md (hunk 1 above)
Comment (SiTaggart) [reply .changeset/thirty-poets-notice.md:2]: Icons is a peer to core, so it doesn't need to be included. Everything else is a dependency or _inside_ of core.

**Review by nkrantz — COMMENTED** [review]
reminder: add titles to new header dark/light icons

**Review by serifluous — APPROVED** [review]

**Review by nkrantz — APPROVED** [review]
i can add them

**Pull request merged**